src/code_transformation/util.jl

```diff
@@ -110,3 +110,52 @@ replace_vararg(typ::SymOrExpr, vararg_info::Tuple) =
         vararg_info[2] == :no_N || vararg_info[2] == :Vararg ?
             replace_body(typ, :(Vararg{$(get_body(typ))})) :
             replace_body(typ, :(Vararg{$(get_body(typ)), $(vararg_info[2])}))
+
+"""
+    parse_kwargs(nt_expr) -> NamedTuple
+
+Accepts an expression containing a `NamedTuple` literal and parses it into a `NamedTuple`
+with expressions as values.
+"""
+function parse_kwargs(nt_expr)
+    if isempty(nt_expr.args)
+        return NamedTuple()
+    end
+
+    first_arg = first(nt_expr.args)
+    if first_arg.head == :parameters
+        return parse_kwargs_parameters(nt_expr)
+    elseif first_arg.head == :(=)
+        return parse_kwargs_tuple(nt_expr)
+    else
+        throw(ArgumentError("Unsupported expression $nt_expr for kwargs;"
+            * " they must be passed as a NamedTuple literal"))
+    end
+end
+
+function parse_kwargs_tuple(tup_expr)
+    nt_names = Tuple(first(ex.args) for ex in tup_expr.args)
+    return NamedTuple{nt_names}(last(ex.args) for ex in tup_expr.args)
+end
```

Comment:
Cleaner: ?
```
nt_names = Tuple(first.(ex.args))
return NamedTuple{nt_names}(last.(ex.args))
```

Reply:
That allocates an array, but mine doesn't.

Reply:
This is true.

Reply:
IMO it isn't the end of the world to allocate an array...

Reply:
I've noticed this is actually different code; I'm iterating over the args of the args.